### ISSUE TITLE
Error when generating typings

### ISSUE THREAD

**Issue by kdelwat** [Neufund/Typechain#7]
I get the following error when trying to generate typings for an ABI:

```
Generating typings...

/usr/lib/node_modules/typechain/dist/abiParser.js:22
    abi.forEach(function (abiPiece) {
        ^

TypeError: abi.forEach is not a function
    at Object.parse (/usr/lib/node_modules/typechain/dist/abiParser.js:22:9)
    at Object.generateSource (/usr/lib/node_modules/typechain/dist/generateSource.js:6:41)
    at /usr/lib/node_modules/typechain/dist/cli.js:19:49
    at Array.forEach (native)
    at Object.<anonymous> (/usr/lib/node_modules/typechain/dist/cli.js:15:9)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)

```
I am using Node version 8.2.1. Let me know if I can provide any further information 😄

**Comment by kdelwat** [comment]
Yes, downloaded from npm. I just updated to the new version and still have the same error unfortunately.

I have attached an example ABI that isn't working (extension renamed to txt so that Github doesn't complain!)

[Auction.txt](https://github.com/Neufund/Typechain/files/1533596/Auction.txt)

**Comment by krzkaczor** [comment]
@kdelwat thanks! Okay, I see the problem: you don't have plain ABI file but the one generated by truffle or something i.e. it is a nested object — real abi is inside key "abi". `Typechain` expects that abi file is a long array (value of "abi" key in your example). This what `solc` generates. 

However, I think that supporting these nested ABI files can be more user friendly and I will add it soon.

**Comment by kdelwat** [comment]
Ah that makes sense, sorry for the mistake on my part - I am using truffle to generate them. Thanks for the help, looking forward to the support.

**Comment by krzkaczor** [comment]
Thanks for reporting! It got fixed in https://github.com/Neufund/Typechain/pull/13 and it's released as `0.0.7`.
